app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php

```diff
@@ -140,21 +150,45 @@ private function extractCustomizableOptions(array $cartItemData): array
         return $customizableOptionsData;
     }
 
+    /**
+     * @param $product
+     * @param array $cartItemData
+     * @return array
+     */
+    private function extractDownloadableLinks($product, array $cartItemData): array
+    {
+        $linksData = [];
+
+        if ($product->getLinksPurchasedSeparately()) {
+            $downloadableLinks = $this->arrayManager->get('downloadable_product_links', $cartItemData, []);
```

Comment:
Looks like it's overhead to use `arrayManager` for simple array

Reply:
yes, sir 

**fixed**